### DISCUSSION TITLE
NextJS 12 with typescript decorators?

### DISCUSSION THREAD

**Discussion by meds** [vercel/next.js#30653] [Ideas]
Have just upgraded to NextJS12, everything worked surprisingly well after a few rough edges were sorted (random javascript out of memory exceptions which mysteriously dissapeared when I commented out an unused component!)

I'm currently using a few babel presets so wont be benefiting from the RUST compiler until they can be sorted out, I know styled-components is coming but I wanted to know if there were plans to add support for "@babel/plugin-proposal-decorators",  "@babel/plugin-proposal-class-properties" and "babel-plugin-parameter-decorator"

I use them extensively in my serverless functions with type-graphql (another great package, highly recommended you guys check it out).

**Comment by jozsefsallai** [comment]
I think you should check out #30174 and post your current `.babelrc` config there if you haven't already. I'm confident they will eventually add decorator support, since it does appear like it's a widely requested feature.

**Comment by HerrNiklasRaab** [comment]
well 5 years later we are still here without decorators :)
